File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        int area = 0;
        
        while (left < right) {
            int temp = (right - left) * Math.min(height[left], height[right]);
            area = Math.max(area, temp);
            
            if (height[left] < height[right]) {
                left++;
            } else if (height[right] < height[left]) {
                right--;
            } else {
                left++;
                right--;
            }
                
        }
        
        return area;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int max = 0;
        
        int left = 0;
        int right = height.size()-1;
        
        while (left < right && left < height.size()) {
            int vol = (right - left) * min(height[right], height[left]);
            if (vol > max) max = vol;
            
            if (height[right] < height[left]) {
                right--;
            } else {
                left++;
            }
        }
        
        return max;
    }
};